packages/dashboard-frontend/src/services/bootstrap/index.ts

```diff
@@ -267,4 +271,34 @@ export default class Bootstrap {
     const { requestUserProfile } = UserProfileStore.actionCreators;
     return requestUserProfile()(this.store.dispatch, this.store.getState, undefined);
   }
+
+  private checkInactivityShutdown() {
+    const path = SessionStorageService.remove(SessionStorageKey.ORIGINAL_LOCATION_PATH);
+    if (!path) {
+      return;
+    }
+
+    const state = this.store.getState();
+
+    const workspace = selectAllWorkspaces(state).find(w => w.ideUrl?.includes(path));
+    if (!workspace) {
+      return;
+    }
+
+    if (workspace.isRunning) {
+      window.location.href = workspace.ideUrl!;
+      return;
```

Comment:
This is here as a dashboard side check for https://github.com/eclipse-che/che-dashboard/pull/556#issuecomment-1151572934